src/Audio/Effects/IntoxicatedEffect.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenTK.Audio.OpenAL;
+using RPVoiceChat.Audio;
+using RPVoiceChat.Audio.Effects;
+
+namespace RPVoiceChat.src.Audio.Effects
+{
+    public class IntoxicatedEffect : SoundEffect
+    {
+        protected float toxicRate;
+
+        public IntoxicatedEffect(int source) : base(source) {}
+
+        protected override void GenerateEffect()
+        {
+            float pitch = toxicRate <= 0.2 ? 1 : 1 - (toxicRate / 5);
+            OALW.Source(source, ALSourcef.Pitch, pitch);
```

Comment:
Changing pitch also changes audio playback speed, this proved to be an issue in all of our testing, causing overflow of circular audio buffer. More advanced implementation is required, for now this can be replaced with a comment to avoid accidental usage of this effect

Reply:
Alert added 👍